src/library_pthread.js

```diff
@@ -1242,6 +1242,12 @@ var LibraryPThread = {
       }
     }
   },
+#elif RELOCATABLE
+  // Provide a dummy version of _emscripten_thread_exit_joinable when
+  // RELOCATABLE is used without MAIN_MODULE.  This is because the call
+  // site in pthread_create.c is not able to distinguish between these
+  // two cases.
+  _emscripten_thread_exit_joinable: (thread) => {},
```

Comment:
Perhaps this should error if actually called? I assume it would be invalid to do so?

Reply:
No it can be called on pthread exit.. in any PIC build.  The prolbem is that we only have one single PIC flavor of libc and so we can't distinguish between `-sRELOCATABLE` and `-sMAIN_MODULE` in the native code.